File: ExamPrep/L21/Q1/main.py


mystery1 = (lambda p1, p2, p3: (p1[0] if (len(p3)%2 > 0) else (p2[len(p2)-1])))

print(mystery1("Hello", "World", "Test"))
print(mystery1("CS2613", "Final", "Exam!"))


mystery2 = (lambda v1: (v1 if (v1 < 0) else v1 * -1))
vals = map(mystery2, [4, -2, 14, -13, 3])
for val in vals:
    print(val)
    
mystery3 = (lambda m1: True if (m1 < 40) else False)
filtered = filter(mystery3, [89, 15, 135, 156, 156])
for val in filtered:
    print(val)

File: ExamPrep/L21/Q1/main.js
const mystery1 = (message, fun) => {
    if(message.length > 6){
        let t = message.split(",");
        let s = 0;
        for(let i = 0; i < t.length; i++){
            s = s + parseInt(t[i])
        }
        return s;
    }

    else if(message.length >= 3){
        let m = message.substring(2) + 1 + "hello";
        return m;
    }
    else if(message.length >= 1){
        let m = fun(parseInt(message));
        return m;
    }

    return 0;
}
const mystery2 = (num) => {
    return num * num
}
console.log(mystery1("1,3,7,1,4,9", mystery2))
console.log(mystery1("608", mystery2))
console.log(mystery1("2", mystery2))
    